logstash-core/lib/logstash/plugins/plugin_factory.rb

```diff
@@ -0,0 +1,102 @@
+# encoding: utf-8
+
+module LogStash
+  module Plugins
+
+    class ExecutionContextFactory
+
+      def initialize(agent, pipeline, dlq_writer)
+        @agent = agent
+        @pipeline = pipeline
+        @dlq_writer = dlq_writer
+      end
+
+      def create(id, klass_cfg_name)
+        ExecutionContext.new(@pipeline, @agent, id, klass_cfg_name, @dlq_writer)
+      end
+    end
+
+    class PluginMetricFactory
+
+      def initialize(pipeline_id, metric)
+        @pipeline_id = pipeline_id.to_s.to_sym
+        @metric = metric
+      end
+
+      def create(plugin_type)
+        @metric.namespace([:stats, :pipelines, @pipeline_id, :plugins])
+          .namespace("#{plugin_type}s".to_sym)
+      end
+    end
+
+    class PluginFactory
+      include org.logstash.config.ir.compiler.RubyIntegration::PluginFactory
+
+      def initialize(lir, metric_factory, logger, exec_factory, filter_class)
```

Comment:
any specific reason to pass the filter_class, but not the output_class ?

Reply:
@jakelandis first off, sorry for completely missing that you reviewed this one almost 2 weeks ago. No clue why Github didn't send me an email here :(

As for the question:

The reason for this is that we have two different filter delegator classes for the Ruby and Java exec respectively. For the output delegator Ruby and Java exec use the same code => we must pass the wrapper type to use to the factory.